README.md

```diff
@@ -1 +1,83 @@
 # microcosm-metrics
+
+Opinionated metrics configuration.
+
+Designed primary to support [DataDog statsd](http://docs.datadoghq.com/guides/dogstatsd/), but some
+effort has been made to support [vanilla statsd](https://github.com/etsy/statsd).
+
+
+## Usage
+
+ 1. Configure a metrics client:
+
+        # either use vanilla statsd
+        graph.use("statsd")
+
+        # or use the datadog implementation
+        graph.use("datadog_statsd")
+
+ 2. In either case, use the resulting `graph.metrics` client:
+
+        metrics.increment("foo")
+
+
+## Decorators
+
+The two most common metrics usages are supported via decorator interfaces.
+
+To time function calls, use:
+
+    @graph.metrics_timing("my_func")
+    def my_func():
+        pass
+
+To count function outcomes, use:
+
+    @graph.metrics_counting("my_func")
+    def my_func():
+        pass
+
+For counting, the default behavior is to count function *calls*. To count different outcomes,
+use a custom `Classifier`:
+
+    class TruthyClassifier(Classifier):
+
+        def label_result(self, result):
+            return "truthy" if bool(result) else "falsey"
+
+        def label_error(self, error):
+            return None
+
+Then pass the classifier class to the counting decorator:
+
+    @graph.metrics_counting("my_func", classifier=TruthyClassifier)
+    def my_func():
+        pass
+
+
+## StatsD Testing
```

Comment:
This section is new.

Note that incrementing counters works fine but histograms (timing) do not. I'm not debugging this now.